### PR TITLE
Shift mir-dataflow from `rustc_borrowck` to `rustc_mir` crate.

### PR DESCRIPTION
**Pull request by pnkfelix**
Shift mir-dataflow from `rustc_borrowck` to `rustc_mir` crate.

Turn `elaborate_drops` and `rustc_peek` implementations into MIR passes that also live in `rustc_mir` crate.

Rewire things so `rustc_driver` uses the `ElaborateDrops` from `rustc_mir` crate.

(This PR is another baby step for mir-borrowck; it is a piece of work that other people want to rebase their stuff on top of, namely developers who are doing other dataflow analyses on top of MIR.)

I have deliberately architected this PR in an attempt to minimize the number of actual code changes. The majority of the diff should be little more than changes to mod and use declarations, as well as a few visibility promotions to pub(crate) when a declaration was moved downward in the module hierarchy.

(I have no problem with other PR's that move declarations around to try to clean this up; my goal was to ensure that the diff here was as small as possible, to make the review nearly trivial.)

### CODE REVIEW

**Comment by pnkfelix** [comment]
(I suspect we should not bother landing this until after issue #42903 is resolved, since we may need to revert the `EndRegion` PR, and while the two PR's may be mostly orthogonal, there may be some interactions that would be a pain to tease apart...)

File: src/librustc_mir/transform/rustc_peek.rs [hunk 1]
```diff
@@ -14,12 +14,67 @@ use syntax_pos::Span;
...
+use dataflow::{DefinitelyInitializedLvals, MaybeInitializedLvals, MaybeUninitializedLvals};
+use dataflow::move_paths::{MovePathIndex, LookupResult};
+use dataflow::move_paths::{HasMoveData, MoveData};
+use dataflow;
+
+use dataflow::has_rustc_mir_with;
+
+pub struct SanityCheck;
+
+impl MirPass for SanityCheck {
```
Comment (arielb1) [inline src/librustc_mir/transform/rustc_peek.rs]: nice!

**Review by arielb1 — APPROVED** [review]
r+ after rebase

**Pull request merged**